### PR TITLE
Fix: npm install env #43947

### PR DESCRIPTION
**Pull request by lbeschastny**

### CODE REVIEW

File: lib/utils/process.js [hunk 1]
```diff
@@ -59,7 +61,7 @@ var findEnvVarName = function(env, names) {
 exports.execScript = function(script, options, callback) {
 	var nodeModulesBinPath = path.join(options.cwd, 'node_modules', '.bin');
 
-	var env = _(options.env || {}).defaults(process.env);
+	var env = _({}).defaults(options.env, process.env);
```
Comment (lbeschastny) [inline lib/utils/process.js:64]: Убрал мутацию `options`

File: lib/utils/process.js [hunk 2]
```diff
@@ -11,6 +11,8 @@ exports.exec = function(cmd, options, callback) {
 
 	var spawnOptions = _(options).clone();
 
+	spawnOptions.env = _({}).defaults(options.env, process.env);
```
Comment (lbeschastny) [inline lib/utils/process.js:14]: Not sure if this is a breaking change or not

Comment (okv) [reply lib/utils/process.js:14]: Seems to be a breaking change for me

File: lib/utils/process.js (hunk 1 above)
Comment (okv) [reply lib/utils/process.js:64]: Looks good

**Comment by okv** [comment]
@lbeschastny @RoobinGood  lgtm, but merge is up to you

File: lib/utils/process.js (hunk 2 above)
Comment (RoobinGood) [reply lib/utils/process.js:14]: Yes, looks like breaking change, so it will be 5.0.0

**Pull request merged**